hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1065,41 +1065,91 @@ public function createInsurancePlanTypes() {
   }
 
   /**
-   * Changes Pension Provider to be a contact, instead of an option_value.
+   * Changes Pension Provider to be a contact, instead of an option_value.  All
+   * required changes are wrapped in a transaction and rolled-back if any 
+   * problems arise.
    * 
-   * @return true
+   * @return boolean
+   *   true on success
+   * 
+   * @throws Exception
    */
   function upgrade_1026() {
     $messages = [];
     $pensionsTableName = CRM_Hrjobcontract_BAO_HRJobPension::getTableName();
 
-    $this->createPensionProviderType($messages);
-    $this->replacePensionProviders($pensionsTableName, $messages);
-    $this->removePensionTypeOptionGroup($messages);
-
-    CRM_Core_DAO::executeQuery("
-      ALTER TABLE {$pensionsTableName} 
-      MODIFY pension_type INT(10) unsigned,
-      ADD CONSTRAINT pension_type_contact_id_fk
-      FOREIGN KEY(pension_type)
-      REFERENCES civicrm_contact(id)
-      ON DELETE SET NULL ON UPDATE CASCADE;
-    ");
+    $tx = new CRM_Core_Transaction();
+    try {
+      $this->createPensionProviderType($messages);
+      $this->replacePensionProviders($pensionsTableName, $messages);
+      $this->removePensionTypeOptionGroup($messages);
+      $this->alterPensionsTable($pensionsTableName, $messages);
+    } catch (Exception $ex) {
+      $tx->rollback();
+      throw new Exception('Cannot do upgrade - ' . $ex->getMessage());
+    }
 
     if (!empty($messages) && function_exists('drupal_set_message')) {
       foreach ($messages as $message) {
         drupal_set_message($message);
       }
     }
 
-    return true;
+    return false;
+  }
+
+  /**
+   * Alters pension_type column to be an unsigned integer and adds a foreign
+   * key referencing civicrm_contact.
+   * 
+   * @param type $pensionsTableName
+   *   Name of pension data table
+   * @param type $messages
+   *   Array of messages for salvageable errors found
+   * 
+   * @throws Exception
+   */
+  function alterPensionsTable($pensionsTableName, &$messages) {
+    try {
+      CRM_Core_DAO::executeQuery("
+        ALTER TABLE $pensionsTableName
+        MODIFY pension_type INT(10) unsigned,
+        ADD CONSTRAINT pension_type_contact_id_fk
+        FOREIGN KEY(pension_type)
+        REFERENCES civicrm_contact(id)
+        ON DELETE SET NULL ON UPDATE CASCADE;
+      ");
+    } catch (Exception $e) {
+      $civi_db_settings = parse_url(CIVICRM_DSN);
```

Comment:
for some variables you're using snake_case, but for other you're using camelCase. Can you please use camelCase for everything?